src/model/fxQuotes.js

```diff
@@ -65,13 +66,88 @@ class FxQuotesModel {
     }
   }
 
+  async checkDuplicateFxQuoteRequest (fxQuoteRequest) {
+    try {
+      // calculate a SHA-256 of the request
+      const hash = calculateRequestHash(fxQuoteRequest)
+      this.log.debug(`Calculated sha256 hash of quote request with id ${fxQuoteRequest.conversionRequestId} as: ${hash}`)
+
+      const dupchk = await this.db.getFxQuoteDuplicateCheck(fxQuoteRequest.conversionRequestId)
+      this.log.debug(`DB query for quote duplicate check with id ${fxQuoteRequest.conversionRequestId} returned: ${util.inspect(dupchk)}`)
+
+      if (!dupchk) {
+        // no existing record for this quoteId found
+        return {
+          isResend: false,
+          isDuplicateId: false
+        }
+      }
+
+      if (dupchk.hash === hash) {
+        // hash matches, this is a resend
+        return {
+          isResend: true,
+          isDuplicateId: true
+        }
+      }
+
+      // if we get here then this is a duplicate id but not a resend e.g. hashes dont match.
+      return {
+        isResend: false,
+        isDuplicateId: true
+      }
+    } catch (err) {
+      // internal-error
+      this.log.error(`Error in checkDuplicateFxQuoteRequest: ${getStackOrInspect(err)}`)
+      throw ErrorHandler.ReformatFSPIOPError(err)
+    }
+  }
+
+  async checkDuplicateFxQuoteResponse (conversionRequestId, fxQuoteResponse) {
+    try {
+      // calculate a SHA-256 of the request
+      const hash = calculateRequestHash(fxQuoteResponse)
+      this.log.debug(`Calculated sha256 hash of quote response with id ${conversionRequestId} as: ${hash}`)
+
+      const dupchk = await this.db.getFxQuoteResponseDuplicateCheck(conversionRequestId)
+      this.log.debug(`DB query for quote response duplicate check with id ${conversionRequestId} returned: ${util.inspect(dupchk)}`)
```

Comment:
```
this.log.debug('DB query for fxQuote response duplicate check returned: ', { conversionRequestId,  dupchk }) 
```

Or we we don't want to repeat `conversionRequestId` in all logs across the the method, we can create a child logger an the beginning, and use it through the rest of the method code:
```
async checkDuplicateFxQuoteResponse (conversionRequestId, fxQuoteResponse) {
    const log = this.log.child({ conversionRequestId })
    try {
       ....
       log.debug('DB query for fxQuote response duplicate check returned: ', { dupchk }) 
       //  conversionRequestId will be added to logs output
```

Reply:
done